components/suggestion/command_provider/command_provider.tsx

```diff
@@ -85,22 +83,34 @@ type Props = {
 export type Results = {
     matchedPretext: string;
     terms: string[];
-    items: CommandSuggestionItem[];
+    items: AutocompleteSuggestion[];
     component: React.ElementType;
-};
+}
 
 type ResultsCallback = (results: Results) => void;
 
 export default class CommandProvider extends Provider {
     private isInRHS: boolean;
     private store: Store<GlobalState>;
     private triggerCharacter: string;
+    private appCommandParser: AppCommandParser;
 
     constructor(props: Props) {
         super();
 
         this.store = globalStore;
         this.isInRHS = props.isInRHS;
+        let rootId;
+        let channelId = getCurrentChannelId(this.store.getState());
```

Comment:
(1/5) Is there anywhere that we could imagine that this is used outside of the current channel/RHS thread? I don't generally like accessing the global store more than we have to, and I feel like it could be better to pass in the channel ID and root ID into the provider instead of getting them here. I think we do that for some of the other providers like the at-mention one

Reply:
A problem here is that the `Textbox` component is mounted one time for the main post textbox. When you navigate to a different channel, the same provider is used as the first channel you visited.

There's an issue in `CommandProvider` here. Every time we interact with the `AppCommandParser`, we should be calling its `setChannelContext` method. This is already fixed in the mobile app.

I would think that this issue also affects the `AtMentionProvider` the same way [here](https://github.com/mattermost/mattermost-webapp/blob/20648d714444f9b31ecfc12bce83a464d535a599/components/textbox/textbox.tsx#L80-L89). Do you know how it works correctly per channel, even though it always references the first channel you visited?

https://github.com/mattermost/mattermost-webapp/blob/20648d714444f9b31ecfc12bce83a464d535a599/components/textbox/textbox.tsx#L80-L89

Reply:
It has special handling for the AtMentionProvider that updates its props when the current channel changes

https://github.com/mattermost/mattermost-webapp/blob/20648d714444f9b31ecfc12bce83a464d535a599/components/textbox/textbox.tsx#L118

Ideally, we'd come up with something more generic for this (even possibly going so far as to recreate the Provider on every channel change), but I don't think adding another special case would hurt.

Reply:
That's a little crazy special case code imo. Do you think we'd be improving the command provider by going this route? 2/5 I think it's better to use the global store as it currently is in this case, to avoid the complexity here, but if we are trying to outright avoid using the global store I understand.

FWIW the unit tests for this command provider are using the global store to set the state, and it seems to be working out ok. Also, the command provider was using the global store before this PR (as much as I hate that reasoning).